### ISSUE TITLE
Entries in EXEMPT_SOURCE_MAC stay even after removal

### ISSUE THREAD

**Issue by mav808** [peacey/split-vpn#11]
When I add an entry to EXEMPT_SOURCE_MAC and restart openvpn it is correctly exempted.  However, if I then remove that same entry and kill and restart openvpn it is not removed and stays as exempted when it should now be included.

So, everything is going over VPN, exempt a MAC and the device is exempted, remove that exemption but the device stays exempted.

If I completely restart the UDMP then the device with the MAC that was added and then removed is included in the VPN once more.  It must not be clearing out a route somewhere.

**Comment by peacey** [comment]
Hi Marek,

Are you setting `REMOVE_KILLSWITCH_ON_EXIT=0` ?

**Comment by marekbell** [comment]
I have REMOVE_KILLSWITCH_ON_EXIT=0.

**Comment by peacey** [comment]
If you use that option, you have to run the following commands manually after you kill the script to delete the previous entries. First cd into your VPN config directory, then run the force-down command like so (change tun0 to the correct tun device in your config)

```sh
cd /mnt/data/openvpn/nordvpn
/mnt/data/openvpn/updown.sh tun0 force-down
```

This will delete all entries. Then you can rerun the VPN to bring it up with the new entries.

Here's my reasoning for why I did it this way: When you use the killswitch option and set the script to not remove the entries on exit, I assume that you want to make sure your forced devices will never leak their real IP. If the iptables rules are removed, there is a chance that your real IP could be leaked, even for a millisecond when the old entries are removed and the new entries are added. The only way to fix this would be to keep track of the previous entries and only remove them if the new entries were successfully added. However, this adds quite a bit of complexity to the script and I didn't want that.

So instead for the case where someone uses the killswitch, I left it up to the user to decide when they wish to delete the old entries by running the force-down command as explained above. This means you can safely make sure your devices will not leak their IP first (by turning off your torrent client or whatever first), then running the force-down command when you feel it's safe. Then rerun the VPN script to bring it up with new entries.

If you think there's a better way to handle this, I am open to suggestions!

**Comment by marekbell** [comment]
First off, thank you @peacey for this amazing script.  It is amazingly useful for me and I absolutely love it.  Thank you!

Your explanation makes sense, and I can't suggest anything better for those that care about never even having a millisecond where a packet could leak out.  I completely agree that tracking entries would be far more complexity and work than the benefit it would provide.

Therefore, the only possible improvement I could suggest is one that you have already considered... having an option like  FLUSH_ROUTES_ON_STARTUP to delete the routes to clear them out and start fresh during the script startup; with an appropriate comment to warn the user that enabling that could lead to a momentary leak if enabled.  Basically, for someone like me who does care about traffic stopping if the VPN unexpectedly stops, but is okay with the very small leak risk when starting, then this would simply remove the need for me to remember to manually call the force-down before starting up again.

That said, it's one command so it's no problem at all for me to run now that you've pointed it out, and I can manually add it to the script so I don't have to remember to do it, so if you think this is an edge case then no worries.  And thanks again for this awesome script.

**Comment by peacey** [comment]
Sorry for the delay, @mav808. Let's just keep it with the current way of running force-down. I think that's easy enough to add to your run script as you said, or even run once if you make some change in the configuration. Maybe I'll just make a note in the documentation about forcing down the configuration when making a change with the `REMOVE_KILLSWITCH_ON_EXIT=0` option. 

Thanks for the feedback! I'll be closing this now but feel free to open another issue if you need to.
